scripts/linters/python_linter.py

```diff
@@ -115,20 +117,127 @@ def _check_import_order(self):
 
+    def _check_all_job_listed_in_job_registry_file(self):
+        """This function is used to check that all the one-off and audit jobs
+        are registered in jobs_registry.py file.
+        """
+        def _get_jobs_class_names_in_filepath(filepath, base_class_name):
+            """Returns a list of job class names in the given filepath which has
+            the given base class.
+
+            Args:
+                filepath: str. The filepath of the jobs.
+                base_class_name: str. The name of the base class.
+
+            Returns:
+                list(str). A list of class names which exist in the given
+                filepath delivered from the base class.
+            """
+            class_names = []
+            module_path = filepath.replace('/', '.')[:-3]
+            python_module = importlib.import_module(module_path)
+            for name, clazz in inspect.getmembers(
+                    python_module, predicate=inspect.isclass):
+                all_base_classes = [base_class.__name__ for base_class in
+                                    (inspect.getmro(clazz))]
+                # Check that it's a subclass of 'BaseMapReduceOneOffJobManager'.
+                if base_class_name in all_base_classes:
+                    class_names.append(name)
+            return class_names
+
+        if self.verbose_mode_enabled:
+            python_utils.PRINT('Starting job registry checks')
+            python_utils.PRINT('----------------------------------------')
+        summary_messages = []
+        failed = False
+        jobs_in_cron = [
+            'DashboardStatsOneOffJob',
+            'UserDeletionOneOffJob',
+            'UserQueryOneOffJob',
+            'VerifyUserDeletionOneOffJob'
+        ]
+
+        jobs_registry = importlib.import_module('core.jobs_registry')
+        expected_one_off_jobs_set = set([
+            jobs.__name__ for jobs in jobs_registry.ONE_OFF_JOB_MANAGERS])
+        expected_validation_jobs_set = set([
+            jobs.__name__ for jobs in jobs_registry.AUDIT_JOB_MANAGERS])
+
+        one_off_jobs_list = []
+        validation_jobs_list = []
+        for filepath in self.all_filepaths:
+            if filepath.endswith('prod_validation_jobs_one_off.py'):
+                validation_jobs_list.extend(_get_jobs_class_names_in_filepath(
+                    filepath, 'ProdValidationAuditOneOffJob'))
+            elif filepath.endswith('_jobs_one_off.py'):
+                one_off_jobs_list.extend(_get_jobs_class_names_in_filepath(
+                    filepath, 'BaseMapReduceOneOffJobManager'))
+
+        # Removing jobs which are used in cron.
+        one_off_jobs_list = [
+            job for job in one_off_jobs_list if job not in jobs_in_cron]
+        one_off_jobs_set = set(one_off_jobs_list)
+        if len(one_off_jobs_list) != len(one_off_jobs_set):
+            failed = True
+            summary_message = 'Found one-off jobs with duplicate names'
```

Comment:
What are their names? Be specific, make things easy for the user.

Reply:
Yeah, that's what I was expecting to implement. Fixed!